src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,101 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double ARRAY_SCALE_FACTOR = 1.5;
+    private int size;
+    private Object[] elements;
+
+    public ArrayList(int capacity) {
+        this.elements = new Object[capacity];
+    }
```

Comment:
why do you need this constructor?  is it a good idea to provide such a constructor to the user?